FaceDetectionServer/client/multi_client.py

```diff
@@ -0,0 +1,437 @@
+"""
+Client script that can use multiple communication methods to send images to the
+computervision backend and measure latency.
+"""
+import sys
+import os
+import platform
+import requests
+import subprocess
+import re
+import base64
+import socket
+import websocket
+import struct
+import json
+import time
+import logging
+from threading import Thread
+
+WEBSOCKET_OPCODE_BINARY = 0x2
+PING_INTERVAL = 4
+TEST_PASS = False
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(threadName)s - %(levelname)s - %(message)s')
+fh = logging.FileHandler('multi_client.log')
+fh.setLevel(logging.DEBUG)
+fh.setFormatter(formatter)
+logger.addHandler(fh)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+if platform.system() == "Darwin":
+    PING = "/sbin/ping"
+    PING_REGEX = r'round-trip min/avg/max/stddev = (.*)/(.*)/(.*)/(.*) ms'
+else:
+    PING = "/bin/ping"
+    PING_REGEX = r'rtt min/avg/max/mdev = (.*)/(.*)/(.*)/(.*) ms'
+
+class Client:
+    """ Base Client class """
+
+    # Initialize "Grand total" class variables.
+    total_latency_full_process = 0
+    count_latency_full_process = 0
+    total_latency_network_only = 0
+    count_latency_network_only = 0
```

Comment:
For my understanding, what's the difference between full_process and network_only? Full_process includes server processing time?

Reply:
full_process includes sending the image to the server, server processing time, and receiving the results.

Reply:
network_only is either an ICMP ping rtt value, or the time it takes to open a TCP socket. So no file transfer or server processing time involved for that.